### ISSUE TITLE
Why do floating point arguments get wrong result under debugging in Core 3.0?

### ISSUE THREAD

**Issue by zmjack** [dotnet/runtime#1321]
# Scenario

Firstly, create a simple console application which is only include one other function except Main with a floating point argument.

Then, write simple code and set breakpoints as follows:
```C#
    class Program
    {
        public static float GetSingle(float instance) => instance;

        static void Main(string[] args)
        {
[breakpoint]            var ret = GetSingle(3.14f);
            Console.WriteLine(ret);
        }
    }
```

The breakpoints is necessary, that is the culprit for the wrong result.

Run it, wait for the program to stop at the breakpoint, and then continue.

The correct output is 3.14.

I use either Core 2.0 or Core 2.2, everything is fine.
![ncore22](https://user-images.githubusercontent.com/24598247/71532852-6f423380-2930-11ea-889b-1f49d9f3e124.gif)

But I change it to Core 3.0, the result is going wrong.
![ncore30](https://user-images.githubusercontent.com/24598247/71532859-7701d800-2930-11ea-9500-7f891e69fa12.gif)

So, what might cause the problem?

**Comment by jkotas** [comment]
This is difference between the behavior with breakpoint set vs. breakpoint not set.

@sdmaclea You have been looking into this class of issues recently. Is this one a known one?

**Comment by sdmaclea** [comment]
This was fixed by https://github.com/dotnet/coreclr/pull/25958.  It was fixed after 3.0 was released, and should be fixed in 3.1.

So far, we have not backported the fix to 3.0.

**Comment by sdmaclea** [comment]
I'm going to close this. Feel free to reopen if we think there is a strong need to discuss backporting to 3.0.
